### PR TITLE
Caching with room#102

### PR DESCRIPTION
**Pull request by TigreBois**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/musiconnect/RoomDatabaseTest.java [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    @Before
+    public void saveDatabaseContentsBeforeTesting() {
+        roomDb = AppDatabase.getInstance(startPageRule.getActivity().getApplicationContext());
+        musicianDao = roomDb.musicianDao();
+        mExecutor.execute(() -> {
+            allUsers = musicianDao.getAll();
+            musicianDao.nukeTable();
+        });
+    }
+
```
Comment (JulienHu) [inline app/src/androidTest/java/ch/epfl/sdp/musiconnect/RoomDatabaseTest.java:51]: There might be a dialog asking for location permission that can block the screen, I will add a call to click on the alert

File: app/src/main/java/ch/epfl/sdp/musiconnect/MapsActivity.java [hunk 2]
```diff
@@ -64,12 +58,13 @@
 
 public class MapsActivity extends FragmentActivity implements OnMapReadyCallback,
         GoogleMap.OnMarkerClickListener, GoogleMap.OnInfoWindowClickListener, AdapterView.OnItemSelectedListener {
-    private Date timeLastUpdt;
-    private SimpleDateFormat sdf = new SimpleDateFormat("dd-M-yyyy hh:mm:ss");
 
     private DataBase db = new DataBase();
     private DbAdapter Adb = new DbAdapter(db);
```
Comment (JulienHu) [inline app/src/main/java/ch/epfl/sdp/musiconnect/MapsActivity.java:64]: Will be replacing those lines to include MockDatabase

File: app/src/main/java/ch/epfl/sdp/musiconnect/Musician.java [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
 
 /**
  * @author Manuel Pellegrini, EPFL
  */
+@Entity
 public class Musician extends Person implements Performer {
 
     private String videoURL;
-    private Map<Instrument, Level> instruments;
+    public Map<Instrument, Level> instruments;
```
Comment (JulienHu) [inline app/src/main/java/ch/epfl/sdp/musiconnect/Musician.java:17]: Is there not a getter instead?

File: app/src/main/java/ch/epfl/sdp/musiconnect/User.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
  */
 public abstract class User {
 
-    private MyDate joinDate;
+    public MyDate joinDate;
```
Comment (JulienHu) [inline app/src/main/java/ch/epfl/sdp/musiconnect/User.java:10]: If we need to modify, maybe a getter + setter is better

File: app/src/main/java/ch/epfl/sdp/musiconnect/Musician.java (hunk 3 above)
Comment (TigreBois) [reply app/src/main/java/ch/epfl/sdp/musiconnect/Musician.java:17]: room had problem with these fields; making them public ensured it worked. I'll try to see if I can make a getter work

**Pull request merged**